src/oc_erchef/apps/oc_chef_wm/src/chef_wm_universe.erl

```diff
@@ -0,0 +1,100 @@
+%% -*- erlang-indent-level: 4;indent-tabs-mode: nil; fill-column: 92 -*-
+%% ex: ts=4 sw=4 et
+%% @author Thom May <thom@chef.io>
+%% Copyright 2015 Chef Software, Inc. All Rights Reserved.
+%%
+%% This file is provided to you under the Apache License,
+%% Version 2.0 (the "License"); you may not use this file
+%% except in compliance with the License.  You may obtain
+%% a copy of the License at
+%%
+%%   http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing,
+%% software distributed under the License is distributed on an
+%% "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+%% KIND, either express or implied.  See the License for the
+%% specific language governing permissions and limitations
+%% under the License.
+%%
+
+-module(chef_wm_universe).
+
+-include("oc_chef_wm.hrl").
+
+%% Webmachine resource callbacks
+-mixin([{oc_chef_wm_base, [content_types_provided/2,
+                           finish_request/2,
+                           malformed_request/2,
+                           ping/2,
+                           forbidden/2,
+                           is_authorized/2,
+                           service_available/2]}]).
+
+-export([allowed_methods/2,
+        to_json/2 ]).
+
+
+%% chef_wm behavior callbacks
+-behaviour(chef_wm).
+-export([auth_info/2,
+         init/1,
+         init_resource_state/1,
+         malformed_request_message/3,
+         request_type/0,
+         validate_request/3]).
+
+init(Config) ->
+    oc_chef_wm_base:init(?MODULE, Config).
+
+init_resource_state(_Config) ->
+  {ok, #base_state{}}.
+
+request_type() ->
+  "universe".
+
+allowed_methods(Req, State) ->
+    {['GET'], Req, State}.
+
+-spec validate_request(chef_wm:http_verb(), wm_req(), chef_wm:base_state()) ->
+                              {wm_req(), chef_wm:base_state()}.
+validate_request('GET', Req, #base_state{organization_guid = OrgId, server_api_version = ApiVersion} = State) ->
+    {Req, State#base_state{resource_state = #oc_chef_organization{id = OrgId, server_api_version = ApiVersion}}}.
+
+auth_info(Req, State) ->
+    {{container, cookbook}, Req, State}.
+
+to_json(Req, #base_state{chef_db_context = DbContext,
+                         organization_guid = OrgId} = State) ->
+  DependencyList = chef_db:fetch_all_cookbook_version_dependencies(DbContext, OrgId),
```

Comment:
It's unfortunate IMHO that chef_db transforms `{ok, Result}` -> `Result`, since it means we can't match on the `ok` here.  Rather, we will end up passing the error tuple along meaning that a database problem is going result in a stack trace whose issues is `bad generator` down at line 97.  Given the small size of the module, it isn't a huge problem now.    One possibility is to use a case statement here to handle the error.  I'll see if we have a good example of that somewhere.